src/device-api/actions.ts

```diff
@@ -386,3 +388,36 @@ export const updateTarget = async (force: boolean = false) => {
 	);
 	return false;
 };
+
+/**
+ * Get application information for a single-container app, throwing if multicontainer
+ * Used by:
+ * - GET /v1/apps/:appId
+ */
+export const getSingleContainerApp = async (appId: number) => {
+	eventTracker.track('GET app (v1)', { appId });
+	const apps = await applicationManager.getCurrentApps();
+	const app = apps[appId];
+	const service = app?.services?.[0];
+	if (service == null) {
+		throw new NotFoundError('App not found');
```

Comment:
This used to return `400` but is now returning `404` which is more correct, but is an interface change.

Reply:
I think this is probably OK.

Reply:
Reverted to original interface in 4a5c2f352f6bf25763531089f8cf023a6c2394c4, following discussion to maintain all legacy interfaces except string contents